client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/operations/PingResponse.java

```diff
@@ -126,7 +126,7 @@ void processResponse(Codec codec, ByteBuf buf, HeaderDecoder decoder) {
 
       PingResponse build(short status) {
          assert decoderState == 4 : "Invalid decoder state";
-         return new PingResponse(status, version, keyMediaType, valueMediaType, serverOps);
+         return new PingResponse(status, version.choose(serverVersion), keyMediaType, valueMediaType, serverOps);
```

Comment:
I don't know it this could work the same:

```suggestion
ProtocolVersion handshakeVersion =
   // take min( maxClientSupportedVersion, maxServerSupportedVersion )
   (serverVersion == null || serverVersion.compareTo(version) >= 0) ?
   version : serverVersion;

return new PingResponse(status, handshakeVersion, keyMediaType, valueMediaType, serverOps);
```

Reply:
The choose operation I've tried to always use the highest as possible. For a client `C` and server `S`:

* Both can trace -> the maximum version
* Neither can trace -> the maximum version
* `C` can trace and `S` can not -> `S`' version
* `S` can trace and `C` can not -> `C`'s version

I think that originally we were always using the most recent, so I was unsure to fallback only to the oldest.